File: generate-improv.jl
function generateExercise(tList::Vector, numList::Vector)
    str = ""
    for i in 1:length(tList)
        for j in 1:numList[i]
            str *= rand(tList[i]) * "      "
        end
    end
    return str;
end

function main(args)
    numAdj = args[2]
    numTechniques = args[3]
    numEx = args[1]
    adjectivesF = open("database/adj.txt");
    techniquesF = open("database/techniques.txt");

    adjectives = readlines(adjectivesF);
    techniques = readlines(techniquesF);
    typeList = [adjectives, techniques]

    println("")
    println("")
    for i in 1:numEx
        println(generateExercise(typeList, [numAdj, numTechniques]))
        println("")
    end
    println("")
    println("")
    return 1;
end

main([2, 1, 3])

File: generate-improv.py
import numpy as np


def generateExercise(adj, techniques, numAdj, numTechniques):
    line = ""
    for i in range(numAdj):
        line += np.random.choice(adj)
    for j in range(numTechniques):
        line += np.random.choice(techniques)
    return line


def openDatabase(numAdj, numTechniques, numEx):
    adjectivesF = open("database/adj.txt", "r")
    techniquesF = open("database/techniques.txt", "r")

    adjectives = adjectivesF.readlines()
    techniques = techniquesF.readlines()

    print("\n")
    print("\n")

    for i in range(numEx):
        print(generateExercise(adjectives, techniques, numAdj, numTechniques))
        print("\n")

    print("\n")
    print("\n")


if __name__ == "__main__":
    openDatabase(1, 2, 3)
